manager/src/main/java/io/gingersnapproject/search/opensearch/CommandResponseListener.java

```diff
@@ -0,0 +1,43 @@
+package io.gingersnapproject.search.opensearch;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.ResponseListener;
+
+class CommandResponseListener implements ResponseListener {
+
+   final CompletableFuture<String> future = new CompletableFuture<>();
+
+   @Override
+   public void onSuccess(Response response) {
+      try {
+         String entity = EntityUtils.toString(response.getEntity());
+         future.complete(entity);
+      } catch (IOException exception) {
+         // in this case the command has been executed server side,
+         // the only problem here is to get the response message from the server
+         future.complete(exception.getMessage());
+      }
+   }
+
+   @Override
+   public void onFailure(Exception exception) {
+      if (exception instanceof ResponseException) {
+         /*
+          * The client tries to guess what's an error and what's not, but it's too naive.
+          * A 404 on DELETE should be accepted for instance to support idempotency.
+          */
+         future.complete(exception.getMessage());
```

Comment:
As a consumer of `SearchBackend` what am I supposed to do with this? For example:

```java
      SearchBackend sb;
      sb.put("indexName", "key", "value").
            whenComplete((res, t) -> {
               // What todo with res when it is Expection#getMessage ?
            });
```

Reply:
if you compare the `QueryRL` and the `CommandRL` you can see that we tolerate stuff for the latter we don't for the former.
In the context of a command for instance some `ResponseException` are not errors. For the moment I would say to make the future complete (not exceptionally) that is the same we do in Hibernate Search for instance.
I had introduced the `CommandResponse` object to wrap the tolerated exception, maybe we could restore it.
The problem with Elasticsearch is the fact that the driver raises some `ResponseException`s in some cases even if there is no error...

Reply:
Another idea is to use `CommandResponse` as command future response. The problem is that I wanted to propagate this Elasticsearch specificity to other client. Maybe it is better to address it (the problem of not error `ResponseException`) once for all after we've fully integrated the driver. What you think?

Reply:
> In the context of a command for instance some `ResponseException` are not errors.

Any reason that we can't handle the known cases of this in the `OpenSearchBackend` for the specific command implementations? Retrying, or taking some other action, as appropriate.

That way the consumer of `SearchBackend` doesn't need to worry about these implementation details.

Reply:
There could be some cases in which we don’t have any errors and we shouldn’t do anything.
For instance a 404 in case of delete document does not represent necessarily an error.
My idea would be to address all these cases (one by one) after the integration with the driver has been done.
I added a todo.
The main idea of tolerating some exception that does not represent an error that is specific to the ES client is not to propagate this behavior to other search backends.